### DISCUSSION TITLE
Llama3/GGUF models do not load after v1.67 on Windows.

### DISCUSSION THREAD

**Discussion by dmjohn0x** [YellowRoseCx/koboldcpp-rocm#58] [General]
This has been an ongoing problem since 1.68.0, but I had hoped somebody would mention it, as I was fighting not to add 2FA to my account, thus was not allowed to comment or leave feedback. But there has been several releases and not a soul has mentioned it, so I finally gave in to report this problem.

On Windows, no version after 1.67.0 actually loads Llama3/gguf models. It always quickly returns an access violation error on the screen and immediately closes upon loading. Im currently running windows 11 23H2, on a Ryzen 7700X w/ 32GB Ram and a 7900 XT w/ 20GB Vram. All of the 1.67.0 releases run perfectly for me. But every version after that crashes when I go to run a modern model.

**Comment by klop08** [comment]
Same for me, the last Version that worked was koboldcpp_rocm4allV3.exe. Any other version just throws Access violation errors loading any model if I use Hipblas. Probably rocm 5.7 related issues. Maybe "built by deleting the entire stock AMD ROCm 5.7.1 GPU code folder and replacing it with only ROCm-4-All-5.7.1 Tensile Library files before compiling." would help.

**Comment by klop08** [comment]
Update: Probably not a rocm issue. Didn't realize that 6.1 already dropped. Same issue here. But I think the problem could be Windows 11 itself after all. I noticed that after I download it, the Windows Defender Smart Screen issues a warning for the app not being trusted. It works after I allow it once. If I close it and start again, I don't get any warnings anymore and I only get Access violation. 
Eventviewer doesn't say anything, but I would guess something from windows is blocking Koboldcpp from accessing the GPU.

**Reply by dmjohn0x** [reply]
Yeah, I dont get any defender prompts. (Probably because I've deactivated Windows Defender) So I don't actually know what the issue is. Just that launching a model in anything after 1.67.0 results in some access violation error then it closes almost instantaneously. I tried many models, and they all run fine on 1.67.y0 rocm4allv3.exe, but nothing works on 1.68,1.69, or 1.69.1

**Reply by cloak1505** [reply]
Every .exe release is *always* "untrusted". It costs developers money to get certificates to become a "known publisher". The SmartScreen popup just warns you about running an unrecognized app the first time you run it. I don't think "allow .exe from unknown publisher to run but not use GPU" is a thing. Works for me.

**Reply by klop08** [reply]
I'm aware of the "untrusted" warning. As I mentioned, it works after I allow it, but only once. If I start the .exe file again, I don't get any warning anymore, but I only encounter Access Violation issues after that. If I download it again, I get the warning again, and it works again only once. It's very weird, and maybe there is a correlation. But its only a guess. 

What OS and patch are you using? Are you also on Windows 11 23H2?

**Comment by klop08** [comment]
1.70 also doesn't work properly. Even running it via Python in Windows gives me the same access violation at 0x0000000 loading any model. The .exe at least works once after the download. Still don't know what the issue is.

**Comment by dmjohn0x** [comment]
Can confirm. 1.70 yields the same violation errors loading models in Win11 23H2 as all previous models going back to v1.68.

1.67.0 Rocm4Allv3 was the last version to work on Windows.

**Comment by RachaelsDen** [comment]
The same here. I noticed some support for ROCM under WSL now, so I tried to run it that way as well. While I didn't get the access violation message, it would just freeze, forcing me to kill the process.

I'm using a 7900 XTX if that helps any. I have tried using DDU to remove and reinstall my GPU drivers, but that didn't seem to help. I was wondering if trying the pro drivers might make any difference.

Little added info, this was with the most recent AMD drivers as of today (7/21/2024) , 24.7.1 on Windows 11 Pro  23H2

**Reply by RachaelsDen** [reply]
Will add that compiling from sourcing using ROCm 5.7 and 6.1 both result in the same error. I'll keep digging.

**Comment by Krxon** [comment]


**Reply by Tatalebuj** [reply]
I'm no expert or skilled technician, but I'm using KoboldcCPP_ROCM.exe v 1.70 on Windows 11 Pro (23H2) using an 7900xt.  Here's a screenshot if that helps:  

![image](https://github.com/user-attachments/assets/fafd6b17-9e4c-43e8-bcbd-c83a73e9647f)



If you need any other info or if I can help troubleshoot, let me know.  But be aware I'm not online exceedingly often.  CHeers and goodluck.

**Reply by Tatalebuj** [reply]
![image](https://github.com/user-attachments/assets/5fd93b03-7f59-4b63-b184-4a21a54dcc5c)

**Reply by klop08** [reply]
Well you are using Vulkan. The Issues mentiond only apply on Rocm.  You probably also can test it and see it yourself if you chose the Rocm option.  Vulkan works fine, but is way slowler in my experience.

**Reply by Tatalebuj** [reply]
Oh, I didn't realize that.  I'll go try the 1.67 version with ROCM to see what the difference is.  Thanks.

**Reply by Tatalebuj** [reply]
> Well you are using Vulkan. The Issues mentiond only apply on Rocm. You probably also can test it and see it yourself if you chose the Rocm option. Vulkan works fine, but is way slowler in my experience.  

I was able to run the same GGUF model using 1.70 and ROCM.  

![image](https://github.com/user-attachments/assets/b88336d6-6480-4e33-b369-49a925233340)

![image](https://github.com/user-attachments/assets/913b42d2-7ad8-4e60-9b76-09941efb026b)


Does that help?

**Reply by dmjohn0x** [reply]
1.71 Still doesnt work on Windows. 1.67.yr0 is the last version working on windows, Tatalebuj is likely using Linux and didnt read the title or failed to mention it. YellowRoseCX has yet to acknowledge that versions after 1.67 no longer run on windows. Likely the same issue we were having prior to 1.67 when several releases wouldnt run on Windows.

**Reply by Tatalebuj** [reply]
> 1.71 Still doesnt work on Windows. 1.67.yr0 is the last version working on windows, 

I attempted to load WSL once, but I am pretty sure I've done an upgrade since and swapped hard drives, but maybe?  Could those files I installed help run the windows version of Koboldcpp_ROCM.exe?  Happy to look where ever to help figure that out, I'm just not familiar with how to do so.

>Tatalebuj is likely using Linux and didnt read the title or failed to mention it.

I'm using Windows Pro 11.  And I'm running the koboldcpp_rocm.exe file out of my KoboldCPP_ROCM folder.  I download the latest .exe from YellowRose, and the one I'm currently using says it is 1.70 and here is my latest run using a new .GGUF model.  I'll attach what the txt says on my screen (I tried to paste, but the text got very ugly and I couldn't figure out how to properly format it.)
[ScrnShot.txt](https://github.com/user-attachments/files/16416014/ScrnShot.txt)




>YellowRoseCX has yet to acknowledge that versions after 1.67 no longer run on windows. Likely the same issue we were having prior to 1.67 when several releases wouldnt run on Windows.

I don't know what that is, but I'm happy to learn that I'm somehow using Linux, because I have no capability with Linux (I don't know hardly anything about it and do not remember installing it specifically.)  I do have several other Local LM applications loaded that I used for other purposes from time to time - though I've not found a way to consistently get the systems working, so I'm really not very good at this - so if running OLLAMA or Docker Dekstop could cause me to run KoboldCPP_ROCM in Linux, then maybe you are right.

**Comment by nnyanpasu** [comment]
strange, but my llama3 launches normally on win11 with the latest updates and drivers. In any case, I'm here to ask how you are with launching llama3.1? I haven't experimented with the settings yet, but 3.1 always crashes.
.\KoboldCPP-v1.70.yr0-ROCm.exe --usecublas --gpulayers 55 --contextsize 16384 .\L3-Umbral-Mind-RP-v3.0-8B.Q8_0.gguf

**Reply by Tatalebuj** [reply]
Downloading the model now, will test it out and let you know.

**Reply by Tatalebuj** [reply]
Seems to work for me.  Here are the screenshots.  I just run Koboldccp_rocm.exe and then left the default selection for ROCM.  Changed by device to #2, as #1 doesn't actually use my video card but the integrated GPU, then change the context to whatever will fit on my 7900xt, uncheck the Launch the browser option, and then run it.

![image](https://github.com/user-attachments/assets/74e5d85d-7437-459e-ac40-fa00b02ef452)
![image](https://github.com/user-attachments/assets/df8b33b4-68d7-4d90-998a-4ac23b6ef6e4)
![image](https://github.com/user-attachments/assets/0a05533e-8713-43ef-badc-e20e62c4b2cb)

Happy to answer any other questions.  Cheers,

**Reply by nnyanpasu** [reply]
> Downloading the model now, will test it out and let you know.

> L3-Umbral-Mind-RP-v3.0-8B.Q8_0.gguf is a llama3 based,
btw even lm studio dont support right not rocm+llama 3.1, so I don't expect the current build KoboldCPP 1.71 for amd for win11 gonna work, heh.

**Reply by Krxon** [reply]
> Happy to answer any other questions. Cheers,

I do the same and it fails.
I either run koboldcpp_rocm.exe and load model, all defaults 
or launch CMD and run koboldcpp_rocm.exe 
All defaults. In both ways it fails:
![Failed](https://github.com/user-attachments/assets/c4ab9ffd-71ee-40ce-a4e9-32ad84c1a06d)

**Reply by Tatalebuj** [reply]
You know....your screenshot reminded me of something.  I was running into a similar problem for a while, and then I realized that while I had been updating the Koboldcpp_rocm.exe with each new iteration, I had not gone back and downloaded all of the other files/folders for quite awhile.  So I opened up my Github desktop instance and updated the entire folder, and now it works.  

I hope that helps, because it's all I can think of that's different.

Good luck!

**Reply by dmjohn0x** [reply]
Im currently running Windows 11 x64 23H2, I have ROCm 5.7 and 6.1.0 WinHIP SDKs installed, Im running a 7900XT GPU with the latest drivers (24.7.1 07/15/24) and I have python 3.10 as requested installed. I've not been able to get a model aboved 1.67 to load anything. There are no logs, annoyingly, so trying to share/troubleshoot the issue is a matter of trying to screenshot the window before it closes on me. Im not the only one here who has been claiming 1.68+ doesnt work either. So its stranger to me that you somehow have it running in windows than it is that nobody else has. 

Update: Mother-of-suck, now 1.67 doesnt even work anymore. All I did was install WinHIP SDK 6.1.0, then suddenly 1.67 would crash on everything. Since this didnt even make 1.71 work, I went ahead and uninstalled it, then restarded, but 1.67 still crashes on me. So Im throwing my hands in the air here. I don't know what the fuck is going on and I dont want to re-install windows again just to make this work.

**Reply by dmjohn0x** [reply]
So, it turns out, KoboldCPP is detecting and listing my GPU as display device 3? But when I select 3, it defaults back to my on-board graphics, while display device 2, which has no label, is actually my 7900XT. So with 6.1.0 uninstalled, I can once again run 1.67 on Windows 11. However 1.71 does NOT run at all and yields the following error: 
![image](https://github.com/user-attachments/assets/e770f19e-f6da-4a06-ab2a-b2c5140abd9d)

**Reply by dmjohn0x** [reply]
In the screenshot, I noticed you were using 1.70, not 1.71, so I gave that a try too: 
![image](https://github.com/user-attachments/assets/c78b0830-c72d-4bfc-bb64-a4e4959e476f)

**Reply by Krxon** [reply]
Tried 1.71.1.yr0-ROCm (Preset hipBLAS (ROCm)) and it still doesn't work. It fails with same message.
Version 1.67.yr0-ROCm (koboldcpp_rocm4allV3.exe) still works.
Note than I am using standalone exe and run it by clicking on koboldcpp_rocm4allV3.exe file. 
All defaults.
Note2: Preset Vulkan works but it's slower to load, text gen works slower, and image gen uses CPU instead of GPU

Also, I noticed this in release notes: "ROCm 6.1.2 builds coming soon"
Does it mean that ROMm will be working/fixed in upcoming versions?

**Reply by dmjohn0x** [reply]
1.72 doesnt work either. Nothing after 1.67 works except for one person who has no clue as to why its working or what he did to make it work.

**Reply by dmjohn0x** [reply]
> Tried 1.71.1.yr0-ROCm (Preset hipBLAS (ROCm)) and it still doesn't work. It fails with same message. Version 1.67.yr0-ROCm (koboldcpp_rocm4allV3.exe) still works. Note than I am using standalone exe and run it by clicking on koboldcpp_rocm4allV3.exe file. All defaults. Note2: Preset Vulkan works but it's slower to load, text gen works slower, and image gen uses CPU instead of GPU
> 
> Also, I noticed this in release notes: "ROCm 6.1.2 builds coming soon" Does it mean that ROMm will be working/fixed in upcoming versions?

We don't know as we dont know why every version after 1.67 throws the access violation error.... That and YellowRoseCX has never even acknowledged this thread so we dont even know if he's aware that new builds wont run on most Windows machines.

That said, I did try koboldcpp_v1.72.yr0-rocm_6.1.2.exe and it does seem to be working. So maybe future 6.1.2 builds will work as well.

**Reply by klop08** [reply]
Can confirm koboldcpp_v1.72.yr0-rocm_6.1.2.exe finally works again. But  the most recent "normal" Version koboldcpp_v1.72.yr0-rocm, doesn't work. 
So as @dmjohn0x  said hopefully new future 6.1.2 builds will work as well.

**Reply by YellowRoseCx** [reply]
does this version work for you guys? https://github.com/YellowRoseCx/koboldcpp-rocm/releases/tag/v1.73.1.yr1-ROCm

**Reply by klop08** [reply]
`ggml_cuda_init: found 1 ROCm devices:
  Device 0: AMD Radeon RX 7900 XT, compute capability 11.0, VMM: no
Traceback (most recent call last):
  File "koboldcpp.py", line 4881, in <module>
  File "koboldcpp.py", line 4526, in main
  File "koboldcpp.py", line 894, in load_model
OSError: exception: access violation reading 0x0000000000000000
[16960] Failed to execute script 'koboldcpp' due to unhandled exception!`

No, it doesn't. At least for me. (Running it with ROCm 6.1.2, Windows 11 RX 7900XT). Back to using "koboldcpp_v1.72.yr0-rocm_6.1.2".

**Reply by dmjohn0x** [reply]
> does this version work for you guys? https://github.com/YellowRoseCx/koboldcpp-rocm/releases/tag/v1.73.1.yr1-ROCm

No. 1.72.yr0-rocm_6.1.2 Was working. The version that said to ignore... But this new version doesn't run on windows with the 7900XT:

![image](https://github.com/user-attachments/assets/3b8c26f2-1aba-402e-a622-163c8ca74c23)

A strange observation I've made is that almost everyone who is having issues since 1.68 in this thread seems to be specifically running a 7900XT.

**Reply by nnyanpasu** [reply]
> does this version work for you guys? https://github.com/YellowRoseCx/koboldcpp-rocm/releases/tag/v1.73.1.yr1-ROCm

for me - yes, win11 23h2 7900xtx

**Comment by klop08** [comment]
Just an Info. Newest Version "KoboldCPP-v1.75.2.yr1-ROCm" seem to work!

**Reply by dmjohn0x** [reply]
![image](https://github.com/user-attachments/assets/5fa120b5-3320-4871-96d5-4f7da369abfc)
Not working for me on 23H2 Win 11

**Comment by dmjohn0x** [comment]
I still cannot run any version higher than "koboldcpp_v1.72.yr0-rocm_6.1.2.exe" on my 7900XT... I dont understand why only the version you've tagged as "IGNORE" is the only version that has actually worked since 1.67.yr0_rocm4allV3.exe...

If you have any reasoning for this or know what I can do to use newer versions, I'd be incredibly appreciative.

**Reply by YellowRoseCx** [reply]
I'll try building the newest version differently to see if it helps 7900 cards

**Reply by nnyanpasu** [reply]
just for the note, i'm using 7900xtx and all the latest 5 versions run without problems
win11 24h1, 24.8.1 driver.

**Reply by dmjohn0x** [reply]
Going through comments on this thread makes it seem like this issue mostly seems to be with users who have the 7900XT specifically... I mean, the sample size is relatively small, but everyone here having the same problems as me seem to be using a 7900XT.

**Reply by Krxon** [reply]
> I'll try building the newest version differently to see if it helps 7900 cards

I gave up on this and removed my threads. Spent too much time on this anyway. **Latest version KoboldCPP-v1.75.2.yr1-ROCm wasn't working, Than it suddenly worked for me fine for 2 days and 3rd day it stopped working with same error. Nothing updated, nothing changed. TF!**
By working I mean that it uses rocm and gpu for both language and image model. Not cpu. I can run it using vulkan but when using vulkan, images are garbled 
I am saving for overpriced nvidia gpu anyway, until then, I will stop loosing time and nerves because of 7900XT and stick with latest working version koboldcpp_v1.72.yr0-rocm_6.1.2.exe.
Thanks for the efforts though.

**Reply by YellowRoseCx** [reply]
Could you try the latest version please?
 https://github.com/YellowRoseCx/koboldcpp-rocm/releases/tag/v1.76.yr0-ROCm

**Reply by Krxon** [reply]
> Could you try the latest version please? https://github.com/YellowRoseCx/koboldcpp-rocm/releases/tag/v1.76.yr0-ROCm

Traceback (most recent call last):
  File "koboldcpp.py", line 4937, in <module>
  File "koboldcpp.py", line 4580, in main
  File "koboldcpp.py", line 890, in load_model
OSError: exception: access violation reading 0x0000000000000000
[15648] Failed to execute script 'koboldcpp' due to unhandled exception!

**Reply by YellowRoseCx** [reply]
I'm compiling a test version with 100% stock rocBLAS files straight from the AMD install files to see if it's something with the compiled GPU files being made or something

it'll be done in about 2 hours: https://github.com/YellowRoseCx/koboldcpp-rocm/actions/runs/11318171214/job/31472451213

**Reply by YellowRoseCx** [reply]
if any of you get time to try this out, I've made a new test build:
https://github.com/YellowRoseCx/koboldcpp-rocm/releases/tag/v1.76.yr0-ROCm-7900xt

**Reply by Krxon** [reply]
> if any of you get time to try this out, I've made a new test build: https://github.com/YellowRoseCx/koboldcpp-rocm/releases/tag/v1.76.yr0-ROCm-7900xt

Thanks for the efforts, unfortunately it still crashes:

Microsoft Windows [Version 10.0.22631.4317]
(c) Microsoft Corporation. All rights reserved.

G:\Kobold>koboldcpp_rocm.exe

Welcome to KoboldCpp - Version 1.76.yr0-ROCm
For command line arguments, please refer to --help

Auto Selected HIP Backend...

Auto Recommended GPU Layers: 35
Attempting to use hipBLAS library for faster prompt ingestion. A compatible AMD GPU will be required.
Initializing dynamic library: koboldcpp_hipblas.dll

Namespace(model='', model_param='G:/Kobold/T capybarahermes-2.5-mistral-7b.Q8_0.gguf', port=5001, port_param=5001, host='', launch=True, config=None, threads=11, usecublas=['normal', '0'], usevulkan=None, useclblast=None, usecpu=False, contextsize=4096, gpulayers=35, tensor_split=None, checkforupdates=False, ropeconfig=[0.0, 10000.0], blasbatchsize=512, blasthreads=11, lora=None, noshift=False, nommap=False, usemlock=False, noavx2=False, debugmode=0, onready='', benchmark=None, prompt='', promptlimit=100, multiuser=1, remotetunnel=False, highpriority=False, foreground=False, preloadstory=None, quiet=False, ssl=None, nocertify=False, mmproj=None, password=None, ignoremissing=False, chatcompletionsadapter=None, flashattention=False, quantkv=0, forceversion=0, smartcontext=False, unpack='', nomodel=False, showgui=False, skiplauncher=False, hordemodelname='', hordeworkername='', hordekey='', hordemaxctx=0, hordegenlen=0, sdmodel='', sdthreads=11, sdclamped=0, sdvae='', sdvaeauto=False, sdquant=False, sdlora='', sdloramult=1.0, whispermodel='', hordeconfig=None, sdconfig=None, noblas=False)

Loading model: G:\Kobold\T capybarahermes-2.5-mistral-7b.Q8_0.gguf

The reported GGUF Arch is: llama
Arch Category: 0

Identified as GGUF model: (ver 6)
Attempting to Load...

Using automatic RoPE scaling for GGUF. If the model has custom RoPE settings, they'll be used directly instead!
It means that the RoPE values written above will be replaced by the RoPE values indicated after loading.
System Info: AVX = 1 | AVX_VNNI = 0 | AVX2 = 1 | AVX512 = 0 | AVX512_VBMI = 0 | AVX512_VNNI = 0 | AVX512_BF16 = 0 | FMA = 1 | NEON = 0 | SVE = 0 | ARM_FMA = 0 | F16C = 1 | FP16_VA = 0 | RISCV_VECT = 0 | WASM_SIMD = 0 | BLAS = 1 | SSE3 = 1 | SSSE3 = 1 | VSX = 0 | MATMUL_INT8 = 0 | LLAMAFILE = 0 |
Traceback (most recent call last):
  File "koboldcpp.py", line 4937, in <module>
  File "koboldcpp.py", line 4580, in main
  File "koboldcpp.py", line 890, in load_model
OSError: exception: access violation reading 0x0000000000000000
[11040] Failed to execute script 'koboldcpp' due to unhandled exception!

**Reply by YellowRoseCx** [reply]
have you tried other models? bc at this point, there's nothing wrong with the program or the rocblas files. It seems like only some people with 7900xt might get this error and as you mentioned a few days ago, at one point, it worked for 2 days and then stopped working the 3rd day, so I think it's more something with your computer setup. (I don't mean that to sound rude, but the program wouldn't have changed just sitting on your pc)
       
Since you have a 7900XT, have you tried using WSL?

**Reply by Krxon** [reply]
> Since you have a 7900XT, have you tried using WSL?
Nope. But as I said before, I will not spend any more time on this.
I did testing because you asked :)
Version * Koboldcpp_v1.72.yr0-rocm_6.1.2 *  still works totally fine - most of my gguf models and sd 1.5  models are working with it.
Few other ppl reported this as well, where koboldcpp_v1.72.yr0-rocm_6.1.2.exe is the last one working.
Maybe some newer versions will work again, until then I am saving for 4090
Thanks a lot for the efforts, though :)

**Reply by eo2day** [reply]
Can attest that this is still happening with the latest ROCm build release.
CPU: Ryzen 7 7800X3D 8 Core.
GPU: AMD 7900xt
Latest Windows 11 - 23H2.

I run Comfyui and ForgeUI (SD web ui) just fine.

```
PS D:\LLMs\Kobold> .\koboldcpp_rocm.exe
***
Welcome to KoboldCpp - Version 1.76.yr1-ROCm
For command line arguments, please refer to --help
***
Auto Selected HIP Backend...

Auto Recommended GPU Layers: 43
Attempting to use hipBLAS library for faster prompt ingestion. A compatible AMD GPU will be required.
Initializing dynamic library: koboldcpp_hipblas.dll
==========
Namespace(model='', model_param='D:/LLMs/Kobold/Models/mythalion-kimiko-v2.Q4_K_M.gguf', port=5001, port_param=5001, host='', launch=True, config=None, threads=7, usecublas=['normal', '0'], usevulkan=None, useclblast=None, usecpu=False, contextsize=4096, gpulayers=43, tensor_split=None, checkforupdates=False, ropeconfig=[0.0, 10000.0], blasbatchsize=2048, blasthreads=7, lora=None, noshift=False, nommap=False, usemlock=False, noavx2=False, debugmode=0, onready='', benchmark=None, prompt='', promptlimit=100, multiuser=1, remotetunnel=False, highpriority=False, foreground=False, preloadstory=None, quiet=False, ssl=None, nocertify=False, mmproj=None, password=None, ignoremissing=False, chatcompletionsadapter=None, flashattention=True, quantkv=0, forceversion=0, smartcontext=False, unpack='', nomodel=False, showgui=False, skiplauncher=False, hordemodelname='', hordeworkername='', hordekey='', hordemaxctx=0, hordegenlen=0, sdmodel='', sdthreads=7, sdclamped=0, sdvae='', sdvaeauto=False, sdquant=False, sdlora='', sdloramult=1.0, whispermodel='', hordeconfig=None, sdconfig=None, noblas=False)
==========
Loading model: D:\LLMs\Kobold\Models\mythalion-kimiko-v2.Q4_K_M.gguf

The reported GGUF Arch is: llama
Arch Category: 0

---
Identified as GGUF model: (ver 6)
Attempting to Load...
---
Using automatic RoPE scaling for GGUF. If the model has custom RoPE settings, they'll be used directly instead!
It means that the RoPE values written above will be replaced by the RoPE values indicated after loading.
System Info: AVX = 1 | AVX_VNNI = 0 | AVX2 = 1 | AVX512 = 0 | AVX512_VBMI = 0 | AVX512_VNNI = 0 | AVX512_BF16 = 0 | FMA = 1 | NEON = 0 | SVE = 0 | ARM_FMA = 0 | F16C = 1 | FP16_VA = 0 | RISCV_VECT = 0 | WASM_SIMD = 0 | BLAS = 1 | SSE3 = 1 | SSSE3 = 1 | VSX = 0 | MATMUL_INT8 = 0 | LLAMAFILE = 0 |
Traceback (most recent call last):
  File "koboldcpp.py", line 4937, in <module>
  File "koboldcpp.py", line 4580, in main
  File "koboldcpp.py", line 890, in load_model
OSError: exception: access violation reading 0x0000000000000000
[12428] Failed to execute script 'koboldcpp' due to unhandled exception!
```

**Reply by dmjohn0x** [reply]
> if any of you get time to try this out, I've made a new test build: https://github.com/YellowRoseCx/koboldcpp-rocm/releases/tag/v1.76.yr0-ROCm-7900xt

That doesn't work, nor does 1.78 or the latest 1.79.yr0. I dont know what you did differently for koboldcpp_v1.72.yr0-rocm_6.1.2.exe, but thats the last version I've got that works. And the non 6.1.2.exe version doesnt work. So I'd focus on whatever you did different there.

**Reply by PhialsBasement** [reply]
> > if any of you get time to try this out, I've made a new test build: https://github.com/YellowRoseCx/koboldcpp-rocm/releases/tag/v1.76.yr0-ROCm-7900xt
> 
> That doesn't work, nor does 1.78 or the latest 1.79.yr0. I dont know what you did differently for koboldcpp_v1.72.yr0-rocm_6.1.2.exe, but thats the last version I've got that works. And the non 6.1.2.exe version doesnt work. So I'd focus on whatever you did different there.

Whats your rocm version?

**Comment by zac-market** [comment]
For what it's worth I'm using `v1.77.yr1-ROCm`, and it has been working absolutely beautifully on both my 6800 XT and 7900 XTX.

I'm on Windows 11 Pro 24H2.

**Comment by dmjohn0x** [comment]
![image](https://github.com/user-attachments/assets/16791453-88ae-4d5c-9389-c2475b18bf25)
This screenshot is of the latest 1.81.1 version not running on Windows 11 23H2 with a 7900 XT.
Ive mentioned before koboldcpp_v1.72.yr0-rocm_6.1.2.exe was the last version that worked for me. I even updated versions of VideoCard driver and re-installed ROCm through the new method of WSL2 as outlined on the AMD ROCm git page.

**Reply by dmjohn0x** [reply]
Same thing for version 1.80.3
![image](https://github.com/user-attachments/assets/a310891c-fa4b-49f5-bfa7-b0a71640df12)

**Reply by dmjohn0x** [reply]
And here is 1.72.0 the last working version I've ever managed to get to work. I think before koboldcpp_v1.72.yr0-rocm_6.1.2.exe the last working version was 1.68.0yr0.
![image](https://github.com/user-attachments/assets/9d87808e-d32a-4c69-bb94-8e31907f8368)
![image](https://github.com/user-attachments/assets/af6f54ca-2972-4c76-b054-c4fa0d31f0e6)
I have no idea why none of the latest versions help. And I'd be ever so happy if anyone else could figure this out as im not alone in this thread. Just most others seemed to have either moved on and got an NVidia GPU or swapped to using linux altogether.

**Reply by Krxon** [reply]
I had similar, if not the same problems like you. And the same OS and the same GPU.
What helped me was uninstalling removing all traces of python, git, torch, pytorch, conda, etc, 
Read: ALL stuff needed to install AI stuff, from program files, windows, all folders and even deleting caches from various places in my user folder. After that, most of versions of KoboldCPP works flawlessly, even latest KoboldCPP-v1.80.3.yr0-ROCm
